sim/pkg/plugin/vpa_plugin.go

```diff
@@ -0,0 +1,49 @@
+package plugin
+
+import (
+	"github.com/hashicorp/go-plugin"
+	"github.com/josephburnett/sk-plugin/pkg/skplug"
+	"github.com/josephburnett/sk-plugin/pkg/skplug/proto"
+	"strconv"
+	"sync/atomic"
+)
+
+type vpaPluginPartition struct {
+	partition string
+}
+
+var vpaPartitionSequence int32 = 0
+var vpaPluginServer skplug.Plugin
+var vpaClient *plugin.Client
+
+func NewVpaPluginPartition() PluginPartition {
```

Comment:
I see that you're creating two plugin hosts, one for HPA and one for VPA.  However if each plugin provides its capabilities, you don't need to differentiate.  You have have just one plugin host that takes a *list* of plugins to start.  Then it could call `Capabilities` for each one to find out what callbacks they want.  Then dispatch calls appropriately.

This would mean you don't have an "HPA" or "VPA" plugin.  You would just have 1-N plugins with some callback hooks.  One of the hooks is to provide vertical or horizontal recommendations.  Another hook is to watch metrics.  But there might be plugins which watch metrics and do not provide recommendations.  E.g. a more detailed backend for the UI to drill into simulation runs.  Or an e2e test asserting plugin.

Reply:
See the commit 33551dc565d916ba8311a50ff86fabcfea41b6ad